# pYPK5

Cassette MX4 amplififcação:

 - primers 477 e 531
 - template pAG32.

Cassette 2u amplificação:

 - primers 534 e 535
 - template pYPK0

pYPK0 digerido com StuI.

In [29]:
from pydna.all import *

In [30]:
pYPK0 = read("pYPK0.gb")

In [31]:
githubuser="BjornFJohansson"
gistid = "c5424b7ebbf553c52053"
x=download_text('https://gist.githubusercontent.com/{u}/{gid}/raw'.format(u=githubuser, gid=gistid))
pAG32 =read(x)

In [32]:
pAG32.cseguid()

ttCV-qk-4UzU8BkVhaswwqdfGlg

[pAG32 sequence](https://gist.github.com/BjornFJohansson/c5424b7ebbf553c52053)

In [33]:
p477, p531, p535, p534=parse(''' 
>477_bleMX4r
agtggaacgaaaactcacgttaagggattttggtcatgaggagctcgttttcga

>531_pSU1_MX4f (67-mer)
taattctcatctttgacagcttatcatcgataagtccccgccgggtc

>535_pSU0_2u_Fwd (21-mer)
taactgtcagaccaagtttac

>534_pSU0_2u_rev (27-mer)
ttatcgatgataagctgtcaaagatga''')

In [34]:
hygmarker = pcr(p477, p531, pAG32)

In [35]:
two_micron =pcr(p534, p535, pYPK0)

In [36]:
from Bio.Restriction import StuI

In [37]:
vect, stuffer = pYPK0.cut(StuI)

In [38]:
asm=Assembly([hygmarker, two_micron, vect])

In [39]:
asm

Assembly:
Sequences........................: [1721] [1619] [4982]
Sequences with shared homologies.: [1721] [1619] [1619] [1721] [4982] [4982]
Homology limit (bp)..............: 25
Number of overlaps...............: 9
Nodes in graph(incl. 5' & 3')....: 8
Only terminal overlaps...........: No
Circular products................: [7219] [6299]
Linear products..................: [7804] [6784] [6332] [6218] [6016] [5902] [5096] [3307] [2539] [2425] [1505] [485] [33]

In [57]:
x,y = asm.circular_products
(x,y)

(Contig(o7219), Contig(o6299))

The larger recombination product x 7219 contains the 
pYPK0 vector two times in a row
the product we want is the y sequence 6299bp

    x:
     -|1619bp_PCR_prod|1505
    |                  \/
    |                  /\
    |                  1505|pYPK0|66
    |                             \/
    |                             /\
    |                             66|pYPK0|41
    |                                      \/
    |                                      /\
    |                                      41|1721bp_PCR_pr_rc|33
    |                                                          \/
    |                                                          /\
    |                                                          33-
    |                                                             |
     -------------------------------------------------------------


    y:
     -|1619bp_PCR_prod|66
    |                  \/
    |                  /\
    |                  66|pYPK0|41
    |                           \/
    |                           /\
    |                           41|1721bp_PCR_pr_rc|33
    |                                               \/
    |                                               /\
    |                                               33-
    |                                                  |
     --------------------------------------------------

In [43]:
y.figure()

 -|1619bp_PCR_prod|66
|                  \/
|                  /\
|                  66|pYKP0|41
|                           \/
|                           /\
|                           41|1721bp_PCR_pr_rc|33
|                                               \/
|                                               /\
|                                               33-
|                                                  |
 --------------------------------------------------

In [44]:
pYPK5 = y.synced(pYPK0)

In [45]:
pYPK5.cseguid()

bYXaKaxDTMFN5_B-QWoDqATVAb0

In [46]:
pYPK5.locus = "pYPK5"
pYPK5.stamp()

cSEGUID_bYXaKaxDTMFN5_B-QWoDqATVAb0

In [47]:
pYPK5.write("pYPK5.gb")

pYPK5.gb

[DOWNLOAD](pYPK5.gb)

In [48]:
from pydna.all import *
r =read("pYPK5.gb")
r.verify_stamp()

cSEGUID_bYXaKaxDTMFN5_B-QWoDqATVAb0